src/EditorFeatures/VisualBasicTest/UseCompoundAssignment/UseCompoundAssignmentTests.vb

```diff
@@ -0,0 +1,366 @@
+"public class C
+    sub M(a as string)
+        a &= 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestExponentiationExpression() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    sub M(a as integer)
+        a [||]= a ^ 10
+    end sub
+end class",
+"public class C
+    sub M(a as integer)
+        a ^= 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestLeftShiftExpression() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    sub M(a as integer)
+        a [||]= a << 10
+    end sub
+end class",
+"public class C
+    sub M(a as integer)
+        a <<= 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestRightShiftExpression() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    sub M(a as integer)
+        a [||]= a >> 10
+    end sub
+end class",
+"public class C
+    sub M(a as integer)
+        a >>= 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestField() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    private a as integer
+
+    sub M()
+        a [||]= a + 10
+    end sub
+end class",
+"public class C
+    private a as integer
+
+    sub M()
+        a += 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestFieldWithThis() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    private a as integer
+
+    sub M()
+        me.a [||]= me.a + 10
+    end sub
+end class",
+"public class C
+    private a as integer
+
+    sub M()
+        me.a += 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestStaticFieldThroughType() As Task
+            Await TestInRegularAndScriptAsync(
+"public class C
+    shared private a as integer
+
+    sub M()
+        C.a [||]= C.a + 10
+    end sub
+end class",
+"public class C
+    shared private a as integer
+
+    sub M()
+        C.a += 10
+    end sub
+end class")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseCompoundAssignment)>
+        Public Async Function TestStaticFieldThroughNamespaceAndType() As Task
+            Await TestInRegularAndScriptAsync(
+"namespace NS
+    public class C
+        shared private a as integer
+
+        sub M()
+            NS.C.a [||]= NS.C.a + 10
```

Comment:
so, we assume property or field doesn't have side effect.

Reply:
oops. ignore field. I meant property.

Reply:
it's more that we require that there only be a single read/write to the property (since that's what will happen once it is translaed to a compound assignment).  So if there's only a single read/write, we offer for a property, otherwise, we don't.